modules/tensor_mechanics/doc/content/modules/tensor_mechanics/1d_elastic_waves.md

```diff
@@ -0,0 +1,107 @@
+Equation (1) can be written as:
+\begin{equation}
+     -\nabla \cdot \sigma + \frac{\partial^2 u}{\partial t^2} = 0 \label{eq3}
+\end{equation}
+
+Now, we consider the plane wave $u= r e^{i(\omega t - k x)}$ is a solution for equation (3), with $r$ being the complex amplitudes, and $k$ is the wave number and $i=\sqrt{-1}$. "hint: you can verify that $r e^{i(\omega t - k x)}$ is a solution by direct substitution in equation (3)", $\Longleftrightarrow$ $k^2= \frac{\omega^2}{c_p^2}$, where $k$ is the wave number, and $c_p=\sqrt \frac{E}{\rho}$ is the longitudinal wave velocity. As we noted before that Young's modulus and the density as constants and equal to unity, thus, $c_p=1$.
+
+Now, equation (2) and the associated BCs. and can be written as,
+\begin{equation}
+     -\nabla \cdot \sigma- \omega^2 u = 0 \label{}
+\end{equation}
+
+\begin{equation}
+    \nabla u= ik \, u \, \qquad at \, x=0,
+    \\
+    u= A \, cos(2 \pi f t) \, \qquad at \, x=L,\label{}
+\end{equation}
+
+Transforming equation (4) and the associated boundary conditions (5) in time to the frequency domain, we obtain:
+\begin{equation}
+     -\nabla \cdot \hat \sigma- \omega^2  U = 0 \label{}
+\end{equation}
+with the following boundary conditions:
+\begin{equation}
+    \nabla U= ik \, U \, \qquad at \, x=0,
+    \\
+U= 0.5 \, \qquad at \, x=L,\label{}
+\end{equation}
+
+where $\hat \sigma = \nabla U$ with $U(x,\omega)$ being the Fourier transform of $u(x,t)$.
+
+!alert note title= Fourier Transfrom of Boundary Conditions
+Not only the pde is transformed to the frequency domain, but also, we must transform all B.Cs as well. Transfroming $A cos(2\pi f t)$ into the frquency domain will result in an impulse at frequency = $f$ with amplitude of A.
+
+When transform the time dependent variable $u$ to the frequency domain, the time-dependency will be omitted. The result is a nother variable $U$ that is frequency dependent. In general, $U$ is complex-valued function/variable in the form of $U_r+iU_i$, where $U_r$ and $U_i$ are the real and the imaginary part of $U$.\\ \\
+At this stage, we split the complex sytem of equations into two $\textbf{real}$ systems, which yields two real-valued systems that live on the same mesh. The strong form PDEs becomes as follows:
+
+\begin{equation}
+    -\nabla \cdot \hat \sigma_r- \omega^2  U_r = 0
+    \\
+    -\nabla \cdot \hat \sigma_i- \omega^2  U_i = 0 \label{}
+\end{equation}
+with the following boundary conditions:
+\begin{equation}
+   \nabla U_r= -k U_i \quad ,\nabla U_i= k U_r\, \qquad at \, x=0,\\
+   U_r=U_i=0 \, \qquad at \, x=L,
+\end{equation}
+
+Note that this decomposition is exact and there is no infromation missed while decompisition. Also note that, as a result of the radiation BCs on left, $U_r$ and $U_i$ are coupled, hence the two systems in (8).
+
+The following is how we define the kernels of this problem,
+
+!listing examples/wave_propagation/1D_elastic_wave_propagation.i block=Kernels
```

Comment:
try:
!listing modules/tensor_mecahnics/examples/wave_propagation/1D_elastic_wave_propagation.i block=Kernels
And same with BCs